packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -77,25 +104,181 @@ function WorklistDataGrid(props) {
+    const cell = grid.current.rows[rowIndex].cells[colIndex];
+    // Remove Tab stop from previous cell in table that has focus and set it to the
+    // cell that was clicked.
+    cell.tabIndex = -1;
+    if (isRowSelectionCell(colIndex) && cell.getElementsByTagName('input').length > 0) {
+      // For row selection cell, the tabstop won't be on the cell itself but the
+      // inner input element so remove it from the input element.
+      cell.getElementsByTagName('input')[0].tabIndex = -1;
+    }
+  };
+
+  const setFocusedRowCol = (newRowIndex, newColIndex, makeActiveElement) => {
+    removeTabStop(focusedRow.current, focusedCol.current);
+    focusedRow.current = newRowIndex;
+    focusedCol.current = newColIndex;
+    setFocus(newRowIndex, newColIndex, makeActiveElement);
+  };
+
+  useEffect(() => {
+    // When row selection mode is turned on or off a row selection column is added or removed.
+    // Therefore, shift the focused cell to the left or right.
+    let newFocusCell = { row: focusedRow.current, col: (focusedCol.current + (hasSelectableRows ? 1 : -1)) };
+
+    if (hasSelectableRows && focusedRow.current === 0 && focusedCol.current === 0) {
+      // When row selection is turned on for the first time, default to the first row selection cell.
+      newFocusCell = { row: 1, col: 0 };
+    } else if (!hasSelectableRows && focusedCol.current === 0) {
+      // When row selection is turned off, if a cell in the row selection had focus, then
+      // refocus on the first cell in that row.
+      newFocusCell = { row: focusedRow.current, col: 0 };
+    }
+    setFocusedRowCol(newFocusCell.row, newFocusCell.col, false);
+    if (currentSelectedCell != null) {
+      setCurrentSelectedCell(null);
+    }
+    // Since the row selection mode has changed, the row selection mode needs to be updated.
+    ariaLiveMsg.current = intl.formatMessage({ id: hasSelectableRows ? 'Terra.worklist-data-grid.row-selection-mode-enabled' : 'Terra.worklist-data-grid.row-selection-mode-disabled' });
+    setRowSelectionModeMessage(ariaLiveMsg.current);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [hasSelectableRows]);
 
-    let nextRow = currentFocusedRow;
-    let nextCol = currentFocusedCol;
+  const isAnyRowSelected = () => (
+    rows.find(r => r.isSelected === true)
+  );
+
+  const handleClearRowSelection = () => {
+    ariaLiveMsg.current = '';
+    if (isAnyRowSelected()) {
+      ariaLiveMsg.current = intl.formatMessage({ id: 'Terra.worklist-data-grid.all-rows-unselected' });
+      // Esc (while in row selection mode and rows are selected): Clear selection
+      if (onClearSelectedRows) {
+        onClearSelectedRows();
+      }
+    } else if (onDisableSelectableRows) {
+      ariaLiveMsg.current = intl.formatMessage({ id: 'Terra.worklist-data-grid.row-selection-mode-disabled' });
+      onDisableSelectableRows();
+    }
+  };
+
+  const selectRows = (selectAllRows, rowId, rowIndex) => {
```

Comment:
How do you unselect all rows?

Reply:
When in row selection mode, Ctrl+A selects all and Esc unselects all selected rows.